src/Strategy.sol

```diff
@@ -134,8 +283,247 @@ contract Strategy is BaseStrategy {
         virtual
         override
         returns (uint256)
+    {}
+
+    // ----------------- SUPPORT FUNCTIONS ----------
+
+    /**
+     * @notice
+     *  Takes the amount in `SYN` and sells it for `want` on solidly
+     * @dev
+     *  Total amount to sell in wei
+     **/
+    function _sellSynToWant(uint256 _amount) internal {
+        _checkAllowance(address(solidlyRouter), address(SYN), _amount);
+
+        IUniswapV2Router02.route[]
+            memory routes = new IUniswapV2Router02.route[](1);
+        routes[0].from = address(SYN);
+        routes[0].to = address(want);
+        routes[0].stable = false;
+        uint256[] memory amounts = solidlyRouter.getAmountsOut(_amount, routes);
+
+        if (amounts[1] > 0) {
+            solidlyRouter.swapExactTokensForTokensSimple(
+                _amount, // amt to sell
+                0, // minimnum amount
+                address(SYN), // token from
+                address(want), // token to
+                false, // stable swap
+                address(this), // to
+                block.timestamp
+            );
+        }
+    }
+
+    function _addliquidity(uint256 _amount) internal {
+        _checkAllowance(address(syn3PoolSwap), address(want), _amount);
+
+        uint256 _expectedLPTokensOut = scaleWantToLP(_amount) *
+            ((ONE_HUNDRED_PERCENT - maxSlippageIn) / ONE_HUNDRED_PERCENT);
+
+        uint256[] memory liquidityToAdd = new uint256[](3);
+        liquidityToAdd[1] = _amount; // USDC
+
+        syn3PoolSwap.addLiquidity(
+            liquidityToAdd,
+            _expectedLPTokensOut,
+            block.timestamp
+        );
+    }
+
+    function _unstakeLPTokens(uint256 _amount) internal {
+        synStakingMC.withdraw(pid, _amount, address(this));
+    }
+
+    function _stakeLPTokens(uint256 _amount) internal {
+        _checkAllowance(address(synStakingMC), address(syn3PoolLP), _amount);
+        synStakingMC.deposit(pid, _amount, address(this));
+    }
+
+    function _minAmountOfLPToWant(uint256 _lpAmount)
+        internal
+        view
+        returns (uint256)
     {
-        // TODO create an accurate price oracle
-        return _amtInWei;
+        uint256 expectedWant = scaleLPToWant(_lpAmount);
+        return
+            expectedWant *
+            ((ONE_HUNDRED_PERCENT - maxSlippageOut) / ONE_HUNDRED_PERCENT);
```

Comment:
this math will not work as expected due to your grouping.

need to change from this
```
expectedWant *
            ((ONE_HUNDRED_PERCENT - maxSlippageOut) / ONE_HUNDRED_PERCENT);
```
to this...

```
expectedWant *
            (ONE_HUNDRED_PERCENT - maxSlippageOut) / ONE_HUNDRED_PERCENT;
```